packages/x-data-grid-premium/src/hooks/features/prompt/useGridPrompt.tsx

```diff
@@ -0,0 +1,126 @@
+  }, [apiRef, columns, rows]);
+
+  const getPromptContext = React.useCallback(
+    (allowDataSampling = false) => {
+      const examples = allowDataSampling ? collectSampleData() : {};
+
+      const columnsContext = columns.map((column) => ({
+        field: column.field,
+        description: column.description ?? null,
+        examples: examples[column.field] ?? column.unstable_examples ?? [],
+        type: column.type ?? 'string',
+        allowedOperators: column.filterOperators?.map((operator) => operator.value) ?? [],
+      }));
+
+      return JSON.stringify(columnsContext);
+    },
+    [columns, collectSampleData],
+  );
+
+  const applyPromptResult = React.useCallback(
+    (result: PromptResponse) => {
+      const interestColumns = [] as string[];
+
+      apiRef.current.setFilterModel({
+        items: result.filters.map((filter, index) => {
+          const item = {
+            id: index,
+            field: filter.column,
+            operator: filter.operator,
+            value: filter.value,
+          };
+
+          const column = columnsLookup[filter.column];
+          if (column.type === 'singleSelect') {
+            const options = getValueOptions(column as GridSingleSelectColDef) ?? [];
+            const found = options.find(
+              (option) => typeof option === 'object' && option.label === filter.value,
+            );
+            if (found) {
+              item.value = (found as any).value;
+            }
+          }
+
+          return item;
+        }),
+        logicOperator: (result.filterOperator as GridLogicOperator) ?? GridLogicOperator.And,
+        quickFilterValues: [],
+      });
+
+      apiRef.current.setRowGroupingModel(result.grouping.map((g) => g.column));
+      apiRef.current.setAggregationModel(result.aggregation);
+
+      apiRef.current.setSortModel(
+        result.sorting.map((s) => ({ field: s.column, sort: s.direction })),
+      );
+
+      const visibleRowsData = getVisibleRows(apiRef);
+      const rowSelectionModel: GridRowSelectionModel = { type: 'include', ids: new Set() };
+      if (result.select !== -1) {
+        for (let i = 0; i < result.select; i += 1) {
+          const row = visibleRowsData.rows[i];
+          const id = apiRef.current.getRowId(row);
+          rowSelectionModel.ids.add(id);
+        }
+      }
+
+      apiRef.current.setRowSelectionModel(rowSelectionModel);
+
+      const targetIndex =
+        Number(columnsLookup[GRID_CHECKBOX_SELECTION_FIELD] !== undefined) +
+        Number(result.grouping.length);
+
+      interestColumns.push(...Object.keys(result.aggregation));
+      interestColumns.push(...result.filters.map((f) => f.column));
+      interestColumns.reverse().forEach((c) => apiRef.current.setColumnIndex(c, targetIndex));
+    },
+    [apiRef, columnsLookup],
+  );
+
+  useGridApiMethod(
+    apiRef,
+    {
+      unstable_getPromptContext: getPromptContext,
+      unstable_applyPromptResult: applyPromptResult,
+    },
```

Comment:
Could possibly use the hierarchical API methods nesting we started with the data source. (We decided to try to do it for new features)
```suggestion
    {
      unstable_aiAssistant: {
        getPromptContext,
        applyPromptResult,
      },
    },
```

Then it would be used like:
```
apiRef.current.unstable_aiAssistant.getPromptContext(...);
```